src/constraints.jl

```diff
@@ -134,8 +134,19 @@ end
 
 Get a constraint's name attribute.
 """
-function name(con_ref::ConstraintRef{<:AbstractModel,<:_MOICON})
-    return MOI.get(con_ref.model, MOI.ConstraintName(), con_ref)::String
+function name(con_ref::ConstraintRef{<:AbstractModel,C}) where {C<:_MOICON}
+    model = owner_model(con_ref)
+    if !MOI.supports(backend(model), MOI.ConstraintName(), C)
```

Comment:
Normally, set says if names can be `set`, not `get`. I guess for read-only model we can make an exception but it's another argument for extending supports (I can't find the issue where we discussed this)

Reply:
If you can't `set`, I don't see how we can expect `get` to return something sensible here.

Reply:
What about `read_from_file` ?

Reply:
That's not relevant at the JuMP level?

Read-only models should define `supports = true`, but throw a `SetAttributeNotAllowed` if `set` is called.

Reply:
Yes I agree, so we should clarify `supports`. We discussed about this but I can't find where. The closest is here: https://github.com/jump-dev/MathOptInterface.jl/issues/1310 but I don't think it's that.

Reply:
https://github.com/jump-dev/MathOptInterface.jl/issues/1628